### ISSUE TITLE
Cannot open First Flyout Item on iOS devices once again

### ISSUE THREAD

**Issue by andreykoniukh** [dotnet/maui#19071]
### Description

Launch app with Shell  main page. I See page for first flyout item from App.Shell.xaml
then Select any other  flyout item. if opens Successfully
then try to open first flyout item once again and nothing happens

### Steps to Reproduce

I used Xaminals app https://learn.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/userinterface-xaminals/
iOS SDK 17
.net 8.0
visual studio version
17.8.0
|Portrait| Landscape|
|:-:|:-:|
|![1](https://github.com/dotnet/maui/assets/9675313/6568612e-8c05-4a02-80ee-252b351990ef)|![2](https://github.com/dotnet/maui/assets/9675313/0a861a81-66cd-42b0-ba2f-ad7b78147c65)|

failed on iOS simulator iphone15 and iOS simulator iphone15 Pro Max and iPhone XR on iOS 16.7 
failed on portrait orientation
works fine on landscape orientation 
Here is my dotnet workload
maui                         8.0.3/8.0.100          SDK 8.0.100
android                    34.0.43/8.0.100       VS 17.8.34309.116, VS 17.4.33213.308
maui-windows          8.0.3/8.0.100          VS 17.8.34309.116, VS 17.4.33213.308
maccatalyst              17.0.8478/8.0.100   VS 17.8.34309.116, VS 17.4.33213.308
ios                            17.0.8478/8.0.100   VS 17.8.34309.116, VS 17.4.33213.308
maui-maccatalyst     8.0.3/8.0.100          VS 17.4.33213.308
maui-ios                   8.0.3/8.0.100          VS 17.4.33213.308
maui-android            8.0.3/8.0.100         VS 17.4.33213.308

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 17

**Comment by drasticactions** [comment]
Trying to port the Forms sample, I couldn't reproduce this. I don't know if you have made changes from the original, or if I did, that may be causing this bug to occur. Could you upload your port so I can cross-check my work?

**Comment by andreykoniukh** [comment]
> https://learn.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/userinterface-xaminals/

Hi you can use Xaminals app thats provided as example of MAUI shell navigation https://learn.microsoft.com/en-us/samples/xamarin/xamarin-forms-samples/userinterface-xaminals/

**Comment by drasticactions** [comment]
Yes, I did. And it worked for me. That's the problem, I did port it and I can't reproduce your error. So if you could upload your version of Xaminals, I can compare my port to yours so I can see if I did something different.

**Comment by andreykoniukh** [comment]
@drasticactions  Here is it.
 https://github.com/andreykoniukh/MYXamainals. also I added my dotnet workload into description

**Comment by mattleibow** [comment]
Might be a duplicate of https://github.com/dotnet/maui/issues/17965

@drasticactions are you able to repro this issue now with the sample?

**Comment by johannes-steurer** [comment]
I exerience the same issue. Also just on iOS

**Comment by andreykoniukh** [comment]
> I exerience the same issue. Also just on iOS

Hi! looks like issue fixed in 8.0.7

**Comment by HarishwaranVijayakumar** [comment]
This issue is not reproducible in the latest .NET MAUI version 9.0.90. I have attached video for your reference. 

https://github.com/user-attachments/assets/00ecc872-1446-44ef-acc9-0eb796f44090
